server/src/main/webapp/WEB-INF/rails/webpack/models/new-environments/environment_agents.ts

```diff
@@ -15,27 +15,45 @@
  */
 
 import Stream from "mithril/stream";
-import {Agent, AgentJSON} from "models/environments/types";
 import {Origin, OriginJSON} from "models/new-environments/origin";
 
+export interface AgentJSON {
+  uuid: string;
+  hostname: string;
+}
+
+export class Agent {
```

Comment:
Can we merge this information with `AgentWithOrigin`? Keeping two models for information of agent seems redundant. Also, we have reused agent class from 'environments/types'. So we can just add hostname in `AgentWithOrigin` class.

Reply:
This is a conscious move as `environments/types` is the older page which will be removed eventually. Hence, wanted to keep the models separate for both pages so that removal is easy. But yes a separate class doesn't makes sense. 
Should I just remove `Agent` class?

Reply:
Yes you can remove Agent class and move `hostname` to `AgentWithOrigin` class